### PR TITLE
Document gradient clipping in Fabric

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Follow up to #16715 

Adds the relevant docs for gradient clipping.

cc @borda @carmocca @justusschock @awaelchli

### CODE REVIEW

**Review by justusschock — APPROVED** [review]

**Review by Borda — APPROVED** [review]

File: docs/source-fabric/api/fabric_methods.rst [hunk 1]
```diff
@@ -61,6 +61,41 @@ This replaces any occurrences of ``loss.backward()`` and makes your code acceler
     fabric.backward(loss)
 
 
+clip_gradients
```
Comment (carmocca) [inline docs/source-fabric/api/fabric_methods.rst:64]: Why don't we use Sphinx's automethod directive and write everything inside the method docstring?

Comment (awaelchli) [reply docs/source-fabric/api/fabric_methods.rst:64]: Because the docstrings are meant for reference, including types etc., while the written docs in the .rst are for user guide. They can be longer and have more examples etc. They may also link to other docs, include images, code blocks etc. In general, one should not mix user guide with docstrings.
